3_cluster/src/gage_year_coverage_by_cluster.R

```diff
@@ -0,0 +1,161 @@
+
+plot_data_coverage <- function(screened_site_list, cluster_table, dv_data_dir, dir_out, estimated_data){
+  #' @description Function to plot the date range for which there is data for each gage. Gages
+  #' are grouped by cluster group.  Will produce a file for each quantile group of clusters.
+  #' 
+  #' @param screened_site_list p1_screened_site_list target. 
+  #' @param cluster_table data frame with ID (gage ID), and any number of columns containing
+  #' cluster id for different groups of quantiles
+  #' @param dv_data_dir directory where daily NWIS streamflow data were saved (currently ./1_fetch/out")
+  #' @param dir_out directory to save plots
+  #' @param estimated_data whether or not to use estimated data in the plots
+  
+  
+  ##there are some sites with more than one discharge field - I think this might have something
+  ##to do with the edits Charlie made when there were more than 1 gage on a single streamline.  For
+  ##now I just removed them - there aren't many and it was breaking my head to try to combine them.
+  
+  df_all <- data.frame()
+  file_list <- list.files(dv_data_dir, pattern = "_dv.csv")
+  screened_site_list <-paste0 (screened_site_list, "_dv.csv") 
+  screened_site_list <- screened_site_list[which(screened_site_list %in% file_list)]
+  
+  df_tally_allgages<-data.frame()
+  for (i in 1:length(screened_site_list)){
+    print(i)
+    
+    fname<- file.path(dv_data_dir,screened_site_list[i])
+    station.ID <- substr(screened_site_list[i], start = 1, stop = 8)
```

Comment:
Will this always work? I think some gages are longer than 8 characters

Reply:
I was worried about that too - but as of now, all our gages have 8 characters.  The reason I had to do this was because there is one gage where the site_no in the csv file has been turned into a date for some reason, so when it gets to that gage, it can't rbind the data to the other sites. So as a work around for now I just extracted the gage number from the file name and replaced the existing site_no column.   I made an issue to look into why that file is messed up. Once that gets fixed, this line could be removed and the existing site_no column could be used.

Reply:
combining those gages is really turning into a pain in the ass, huh?? lol

Reply:
Okay, no problem! We can omit those sites from these plots. Can you add comments above the lines with workarounds so we know why it's programmed that way?